src/smartfin-fw3.ino

```diff
@@ -26,12 +26,22 @@
 SYSTEM_MODE(SEMI_AUTOMATIC);
 SYSTEM_THREAD(ENABLED);
 
-// Statemachine for handeling task-switching
+/**
+ * @brief State machine for handling task switching
+ *
+ * Associates a device state with a corresponding task
+ */
 typedef struct StateMachine_
 {
+    /**
+     * @brief Current device state
```

Comment:
Please look at the usage of this struct in the code.  This struct is not used to keep track of the current state so much as it is used to define the structure of a mapping.